shared/src/main/scala/zeno/examples/MultiPaxosLeader.scala

```diff
@@ -0,0 +1,97 @@
+package zeno.examples
+
+import scala.collection.mutable
+import scala.scalajs.js.annotation._
+import zeno.Actor
+import zeno.Logger
+import zeno.ProtoSerializer
+import zeno.TypedActorClient
+
+@JSExportAll
+object MultiPaxosLeaderInboundSerializer
+extends ProtoSerializer[MultiPaxosLeaderInbound] {
+  type A = MultiPaxosLeaderInbound
+  override def toBytes(x: A): Array[Byte] = super.toBytes(x)
+  override def fromBytes(bytes: Array[Byte]): A = super.fromBytes(bytes)
+  override def toPrettyString(x: A): String = super.toPrettyString(x)
+}
+
+@JSExportAll
+object MultiPaxosLeaderActor {
+  val serializer = MultiPaxosLeaderInboundSerializer
+}
+
+class MultiPaxosLeaderActor[Transport <: zeno.Transport[Transport]](
+  address: Transport#Address,
+  transport: Transport,
+  logger: Logger,
+  config: PaxosConfig[Transport]
+  ) extends Actor(address, transport, logger) {
+    override type InboundMessage = MultiPaxosLeaderInbound
+    override def serializer = MultiPaxosLeaderActor.serializer
+
+    // Verify the Paxos config and get our proposer index.
+    logger.check(config.valid())
+    logger.check(config.proposerAddresses.contains(address))
+    private val index: Int = config.proposerAddresses.indexOf(address)
+
+    // Monotonically increasing (equivalent to round)
+    private var ballotNumber: Double = 0
+
+    // Whether the leader is active
+    private var active: Boolean = false
```

Comment:
Let me know if you need help embedding the leader election stuff into MultiPaxos. I haven't written any examples where one actor is embedded in another, so I don't know how hard it will be.